File: Medium/2025-12-11-3531-Count-Covered-Buildings/solution.java
import java.util.HashSet;
import java.util.Set;

class Solution {
    public int countCoveredBuildings(int n, int[][] buildings) {
        Set<String> buildingSet = new HashSet<>();
        for (int[] building : buildings) {
            buildingSet.add(building[0] + "," + building[1]);
        }

        int coveredCount = 0;

        for (int[] building : buildings) {
            int x = building[0];
            int y = building[1];
            if (buildingSet.contains(x - 1 + "," + y) && 
                buildingSet.contains(x + 1 + "," + y) && 
                buildingSet.contains(x + "," + (y - 1)) && 
                buildingSet.contains(x + "," + (y + 1))) {
                coveredCount++;
            }
        }

        return coveredCount;
    }
}

File: Medium/2025-12-11-3531-Count-Covered-Buildings/solution.py
class Solution:
    def countCoveredBuildings(self, n: int, buildings: List[List[int]]) -> int:
        covered = set()
        building_set = set(map(tuple, buildings))
        
        for x, y in buildings:
            if (x - 1, y) in building_set and (x + 1, y) in building_set and (x, y - 1) in building_set and (x, y + 1) in building_set:
                covered.add((x, y))
        
        return len(covered)